### PR TITLE
Issue #2985: Add object ids by default in uhdm output

### PR DESCRIPTION
**Pull request by hs-apotell**
Issue #2985: Add object ids by default in uhdm output

Also, fixed the regression script update logic to fix line-endings for Windows.

NOTE: PR includes [PR 723](https://github.com/chipsalliance/UHDM/pull/723) from UHDM

### CODE REVIEW

File: scripts/regression.py [hunk 1]
```diff
@@ -745,24 +746,13 @@ def _update_one(params):
...
-      #
-      # with open(golden_log_filepath, 'wt', encoding='cp850') as ostrm:
-      #   for line in lines:
-      #     ostrm.write(line)
-      #     ostrm.write('\n')
-      #   ostrm.flush()
-      #   ostrm.close()
+      # On Windows, fixup the line endings
+      if platform.system() == 'Windows':
+        with open(golden_log_filepath, 'rt', encoding='cp850') as istrm:
```
Comment (hzeller) [inline scripts/regression.py:751]: Wouldn't UTF-8 be more robust in case there is any unicode in the output ? cp850 sounds very 1990ies...

Comment (hs-apotell) [reply scripts/regression.py:751]: That was immediate solution when I saw an encoding issue but that it didn't work. I remember the issue to being only on MSYS2.

That said, the logs have changed quite a bit and I remember you making a specific change to remove non-ascii characters from log.

I will try changing it to utf-8 across the script and see if anything falls. Will follow up with a new PR once the current open PRs are merged and dust settles.

**Comment by alaindargelas** [comment]
This PR and https://github.com/chipsalliance/UHDM/pull/723 do not add-up in functionality.
I reverted https://github.com/chipsalliance/UHDM/pull/723 until there is more evidence that vpiParent info is visible in the visitor.

**Comment by hs-apotell** [comment]
@alaindargelas Targeted change. Will reintroduce the vpiParent change as a separate PR.

**Comment by alaindargelas** [comment]
Actually, this was a bad idea. I'm working on a change in UHDM elaboration, the IDs changes makes the diffs unreadable.

**Pull request merged**